src/pages/admin/absences/[id].tsx

```diff
@@ -1,45 +1,96 @@
+import KintaiAdmin from "@/api/KintaiAdmin";
 import AdminHeader from "@/components/AdminHeader";
 import MyHead from "@/components/MyHead";
 import ShiftAbsenceShow from "@/components/ShiftAbsenceShow";
 import StatusButton from "@/components/StatusButton";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { GetAbsenceForAdmin200ResponseData as Absence } from "@/api/typescript-axios";
 
-// 仮の時間を設定
-const startTime = new Date("5 17, 1998 18:00:00");
-const endTime = new Date("5 17, 1998 23:00:00");
+export default function AdminAbsencesShow() {
+  const router = useRouter();
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [absence, setAbsence] = useState<Absence>();
+  const [updateAbsenceStatus, setUpdateAbsenceStatus] = useState<boolean>();
+  // データフェッチの際、idはNumber型で指定している
+  const id = Number(router.query.id);
 
-const rows = [
-  {
-    detailLabel: "従業員名",
-    detail: "I am 従業員",
-  },
-  {
-    detailLabel: "開始時間",
-    detail: startTime.toLocaleString(),
-  },
-  {
-    detailLabel: "終了時間",
-    detail: endTime.toLocaleString(),
-  },
-  {
-    detailLabel: "承認状況",
-    detail: "未承認",
-  },
-];
+  useEffect(() => {
+    (async () => {
+      // router.isReady はboolean。trueでないと、idがundefinedになってしまうため、trueの時にデータフェッチを行う。
+      if (router.isReady) {
+        try {
+          const absenceRes = await KintaiAdmin.getAbsenceForAdmin(id);
+          setAbsence(absenceRes.data.data);
+        } catch (error) {
+          // TODO: エラー処理と、その共通化
+        }
+        setLoading(false);
+      }
+    })();
+  }, [router.isReady, updateAbsenceStatus]);
```

Comment:
うすうす感づいておられると思いますが、 useEffect を発火させるためだけに updateAbsenceStatus を使うのはアウトです。この記事を読んでみてください。特にこのへん

> 値の変化に反応するためにuseEffectを使うのは良くない

https://zenn.dev/uhyo/articles/useeffect-taught-by-extremist

Reply:
こんな感じで、 fetchAbsence を useEffect の外に出して、他の箇所からも使えるようにすればいいんじゃないでしょうか

```suggestion
  const fetchAbsence = useCallback(async () => {
    // router.isReady はboolean。trueでないと、idがundefinedになってしまうため、trueの時にデータフェッチを行う。
    if (router.isReady) {
      try {
        const id = Number(router.query.id);
        const absenceRes = await KintaiAdmin.getAbsenceForAdmin(id);
        setAbsence(absenceRes.data.data);
      } catch (error) {
        // TODO: エラー処理と、その共通化
      }
      setLoading(false);
    }
  }, [router]);

  useEffect(() => {
    fetchAbsence();
  }, [fetchAbsence]);
```

Reply:
ありがとうございます！
とても学びになります！！

https://github.com/tatata05/next-app/pull/22/commits/64a2394ed7cc3a00a081a8bf08355d2e4ad8c383
修正いたしました。